### ISSUE TITLE
nix-shell doesn't handle double spacing

### ISSUE THREAD

**Issue by puckipedia** [NixOS/nix#9735]
**Describe the bug**
Shebang parsing doesn't handle whitespace well.

**Steps To Reproduce**
1. `#!nix-shell  -p python3`
2. Get the most baffling error ever

**Expected behavior**
Not an incomprehensible error

**`nix-env --version` output**
nix-env (Nix) 2.18.1

**Priorities**

Add :+1: to [issues you find important](https://github.com/NixOS/nix/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc).

**Comment by abathur** [comment]
For the sake of other searchers, what was the baffling error? :)

**Comment by raphaelr** [comment]
```
$ cat test.sh
#!/usr/bin/env nix-shell
#!nix-shell  -p python3

$ ./test.sh
error: syntax error, unexpected ')'

       at «string»:1:107:

            1| {...}@args: with import <nixpkgs> args; (pkgs.runCommandCC or pkgs.runCommand) "shell" { buildInputs = [ ()
             |
              ^

```

**Comment by thufschmitt** [comment]
I can indeed reproduce that on Nix 2.18. But 2.20 works fine, so it seems to have been accidentally fixed in the meantime.

@puckipedia can you try with a more recent Nix, see whether the error gets fixed for you?

**Comment by puckipedia** [comment]
ah yeah, looks like this was patched with 595010b631b0f92ee6eb4afb8494bc9c73e8513b in 2.19
